apps/sms/js/subject_composer.js

```diff
@@ -62,17 +62,6 @@
     }
   }
 
-  /** Without bug 1103011 fixed, subject's caret would not be set at correct
-   * position with dir attribute set to auto. So we need to specify the auto
-   * direction only when subject is not empty.
-   */
-  function updateDirection(value) {
```

Comment:
Please remove unit test that depends on this logc as well [1]. It's failing currently on your PR.

[1] https://github.com/mozilla-b2g/gaia/blob/c79b08251a6d43953f067d9acc185504d4d27142/apps/sms/test/unit/subject_composer_test.js#L212